CoordgenMinimizer.cpp

```diff
@@ -70,12 +74,18 @@ void CoordgenMinimizer::run()
     if (!_interactions.size()) {
         setupInteractions();
     }
-
+    std::vector<float> energy_list(m_maxIterations);
     for (int iterations = 0; iterations < m_maxIterations; ++iterations) {
-        scoreInteractions();
+        energy_list[iterations] = scoreInteractions();
         if (!applyForces(0.1f)) {
             break;
         }
+        if (iterations < 2 * ITERATION_HISTORY_SIZE) {
+            continue;
+        }
+        if (energy_list[iterations - ITERATION_HISTORY_SIZE] - energy_list[iterations] < MAX_NET_ENERGY_CHANGE) {
```

Comment:
This breaks if we go up in energy, are we cool with that?

Reply:
I'm thinking yes, because this cannot be reached until after 200 iterations and all of the examples I have seen are converging by then. There are examples where the energy level hits a valley after 200 iterations, then very slowly increases, and I think we would want to break for those (see the blue line in the second graph I provided).

However I could see it being a problem if we changed step size or `ITERATION_HISTORY_SIZE`. Maybe we should check the absolute value, and assume if the energy increases by a lot, then it is just approaching a peak and will come back down.

Reply:
yeah, I think that's probably so. I just feel silly that this means that we'll almost always terminate minimization at _not_ a minimum, but instead a local oscillation up. As you say, this is usually meaningless. But it feels odd.